### DISCUSSION TITLE
Typo in 02_pytorch_classification.ipynb

### DISCUSSION THREAD

**Discussion by Caesar0714** [mrdbourke/pytorch-deep-learning#1043] [General]
Hi there,

I just want to mention that there is a typo in the part under the question, I think the activation layer is usually put just before the output layer instead of after.
<img width="1302" alt="Screenshot 2024-08-19 at 09 09 24" src="https://github.com/user-attachments/assets/1e32496e-997f-4efc-82e0-b305cd39c0d7">

Best,
Caesar

**Comment by mrdbourke** [comment]
Hey @Caesar0714 ,

That's a great point!

However, the part about "just after the output layer" refers to activation functions such as the `torch.softmax` function being used after a model's output layer to convert output logits to prediction probabilities. 

For example:

```python
outputs = model(x)
prediction_probabilities = torch.softmax(outputs, dim=1)
```

But you are right, you can use non-linear activations throughout the neural network before the output layer as well.

**Reply by Caesar0714** [reply]
Oh, I see. I didn't realise it since the previous code cell puts the ReLU right before the output layer, so I felt a bit confused about the "just after the output layer" in the later content. But anyway, thanks Daniel!
